### PR TITLE
Fix issue 19128 - argument to alloca may be too large

### PR DESCRIPTION
**Pull request by ibuclaw**
Replaces #2258.

### CODE REVIEW

**Comment by ibuclaw** [comment]
Dammit windows, doesn't support basic functionality.
```
Error: cannot mix `core.std.stdlib.alloca()` and exception handling in `_d_arrayassign()`
--- errorlevel 1
```

**Comment by ibuclaw** [comment]
Only fixed `_d_arraysetassign` to not use `scope(exit)`.

For `_d_arrayassign` partially reverted back to use `void[16]` if the element fits, otherwise always use `malloc()`.  This is fine IMO, as this function is deprecated, and the compiler should be calling `_d_arrayassign_l` directly itself.

File: src/rt/arrayassign.d [hunk 1]
```diff
@@ -206,24 +211,20 @@ extern (C) void* _d_arraysetassign(void* p, void* value, int count, TypeInfo ti)
...
-    //Need a temporary buffer tmp[] big enough to hold one element
-    void[16] buf = void;
-    void[] tmp;
-    if (element_size > buf.sizeof)
-    {
-        tmp = alloca(element_size)[0 .. element_size];
-    }
-    else
-        tmp = buf[];
+    // Need a temporary buffer tmp[] big enough to hold one element
```
Comment (ZombineDev) [inline src/rt/arrayassign.d]:
```suggestion
    // Need a temporary buffer big enough to hold one element.
    // Stack allocate if element_size < 512.
```

**Comment by thewilsonator** [comment]
Please give the a kick, FreeBSD seems to be getting a lot of those socket errors.

**Comment by wilzbach** [comment]
> Please give the a kick, FreeBSD seems to be getting a lot of those socket errors.

Auto-tester's agents will automatically restart.

**Comment by wilzbach** [comment]
Though I merged this manually as currently some Buildkite agents are running into timeouts.

**Pull request merged**